baseplate/experiments/providers/__init__.py

```diff
@@ -55,28 +67,30 @@ def parse_experiment(config):
     :return: A subclass of :py:class:`Experiment` for the given experiment
         type.
     """
-    experiment_type = config["type"].lower()
-    experiment_id = config["id"]
+    experiment_type = config.get("type")
+    if experiment_type:
+        experiment_type = experiment_type.lower()
+    experiment_id = config.get("id")
```

Comment:
why change from `["id"]` to `.get("id")`? with the assert on the next line all we've done is change the error from a fairly understandable `KeyError: id` to `AssertionError` in the case where the ID wasn't defined

Reply:
Fair. Altered it to still raise a KeyError (let me know if you think there's a better exception type), but made the error message a bit more explicit.